### ISSUE TITLE
Direct outbond ipv6 connect timeout

### ISSUE THREAD

**Issue by ActiveIce** [SagerNet/sing-box#191]
### Welcome

- [X] Yes, I'm using the latest major release. Only such installations are supported.
- [X] Yes, I'm using the latest Golang release. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, log, etc).

### Description of the problem

I'm using sing-box just to hijack dns and use DoH instead.
When dns.servers.strategy set to "ipv4_only", everything's working correctly.
When dns.servers.strategy set to "prefer_ipv4", I cannot access to domain with ipv6 address (eg. www.bilibili.com) or I can access  only the first 10-20 seconds.
I've check my network and it can access to ipv6.
Since I did not using sing-box as a proxy, there's only client config provided.

### Version of sing-box

<details>

```console
$ sing-box version
sing-box version 1.1-beta11

Environment: go1.19.2 windows/amd64
Revision: f3d1b591736e5c47176a184d54db456f9628690c
CGO: enabled
```

</details>

### Server and client configuration file

<details>

```console
{
  "log": {
    "level": "debug"
  },
  "dns": {
    "servers": [
      {
        "tag": "alidns",
        "address": "https://223.5.5.5/dns-query",
        "strategy": "prefer_ipv4",
        "detour": "direct"
      }
    ]
  },
  "inbounds": [
    {
      "type": "tun",
      "interface_name": "tun0",
      "inet4_address": "10.0.0.0/30",
      "inet6_address": "fd0a::/126",
      "auto_route": true
    }
  ],
  "outbounds": [
    {
      "type": "direct",
      "tag": "direct"
    },
    {
      "type": "dns",
      "tag": "dns-out"
    }
  ],
  "route": {
    "rules": [
      {
        "port": 53,
        "outbound": "dns-out"
      }
    ],
    "auto_detect_interface": true
  }
}

```

</details>

### Server and client log file

<details>

```console
INFO[0105] [2962870022] inbound/tun[0]: inbound packet connection from [fd0a::]:50104
INFO[0105] [2962870022] inbound/tun[0]: inbound packet connection to [fd0a::1]:53
DEBUG[0105] [2962870022] router: match[0] port=53 => dns-out
INFO[0105] [2748979187] inbound/tun[0]: inbound packet connection from [fd0a::]:63188
INFO[0105] [2748979187] inbound/tun[0]: inbound packet connection to [fd0a::1]:53
DEBUG[0105] [2748979187] router: match[0] port=53 => dns-out
DEBUG[0105] [2962870022] dns: exchange s1.hdslb.com. IN A
INFO[0105] [3843120482] inbound/tun[0]: inbound packet connection from [fd0a::]:64226
INFO[0105] [3843120482] inbound/tun[0]: inbound packet connection to [fd0a::1]:53
DEBUG[0105] [3843120482] router: match[0] port=53 => dns-out
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. CNAME s1.hdslb.com. 4 IN CNAME bstatic.hdslb.com.
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. CNAME bstatic.hdslb.com. 4 IN CNAME s1.hdslb.com.04f6a54d.c.cdnhwc1.com.
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. CNAME s1.hdslb.com.04f6a54d.c.cdnhwc1.com. 4 IN CNAME hcdnw.nozj.bili.c.cdnhwc2.com.
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 58.59.128.112
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 58.59.128.110
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 222.216.124.72
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 222.216.124.73
INFO[0105] [927781168] inbound/tun[0]: inbound packet connection from [fd0a::]:57298
INFO[0105] [927781168] inbound/tun[0]: inbound packet connection to [fd0a::1]:53
DEBUG[0105] [927781168] router: match[0] port=53 => dns-out
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 222.216.124.71
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 222.216.124.69
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 58.59.128.113
INFO[0105] [2962870022] dns: exchanged s1.hdslb.com. A hcdnw.nozj.bili.c.cdnhwc2.com. 4 IN A 58.59.128.107
DEBUG[0105] [2748979187] dns: exchange s1.hdslb.com. IN AAAA
INFO[0105] [2748979187] dns: exchanged s1.hdslb.com. CNAME s1.hdslb.com. 1 IN CNAME bstatic.hdslb.com.
INFO[0105] [2748979187] dns: exchanged s1.hdslb.com. CNAME bstatic.hdslb.com. 1 IN CNAME s1.hdslb.com.w.kunlunar.com.
DEBUG[0105] [3843120482] dns: exchange t.bilibili.com. IN A
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. CNAME t.bilibili.com. 1 IN CNAME a.w.bilicdn1.com.
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.144
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 14.17.92.70
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 14.17.92.71
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 14.17.92.72
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 14.17.92.73
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 14.17.92.74
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.228.17
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.228.18
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.228.19
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.228.20
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.228.21
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.232.1
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.232.2
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.232.3
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 59.36.232.4
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.134
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.136
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.138
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.140
INFO[0105] [3843120482] dns: exchanged t.bilibili.com. A a.w.bilicdn1.com. 1 IN A 183.60.220.142
DEBUG[0105] [927781168] dns: exchange t.bilibili.com. IN AAAA
INFO[0105] [927781168] dns: exchanged t.bilibili.com. CNAME t.bilibili.com. 1 IN CNAME a.w.bilicdn1.com.
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::6
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::7
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::8
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::9
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:10:1402:0:1:1:38
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:10:1402:0:1:1:39
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::4
INFO[0105] [927781168] dns: exchanged t.bilibili.com. AAAA a.w.bilicdn1.com. 1 IN AAAA 240e:97d:2014:204::5
INFO[0105] [4174103055] inbound/tun[0]: inbound connection from [fd0a::]:55365
INFO[0105] [4174103055] inbound/tun[0]: inbound connection to [240e:97d:2014:204::6]:443
INFO[0105] [4174103055] outbound/direct[direct]: outbound connection to [240e:97d:2014:204::6]:443
INFO[0107] [1487056419] inbound/tun[0]: inbound connection from [fd0a::]:55367
INFO[0107] [1487056419] inbound/tun[0]: inbound connection to [2a01:111:2003::52]:80
INFO[0107] [1487056419] outbound/direct[direct]: outbound connection to [2a01:111:2003::52]:80
DEBUG[0110] [4174103055] inbound/tun[0]: connection closed: dial tcp [240e:97d:2014:204::6]:443: i/o timeout
INFO[0110] [4082274589] inbound/tun[0]: inbound connection from [fd0a::]:55370
INFO[0110] [4082274589] inbound/tun[0]: inbound connection to [240e:97d:2014:204::6]:443
INFO[0110] [4082274589] outbound/direct[direct]: outbound connection to [240e:97d:2014:204::6]:443
DEBUG[0112] [1487056419] inbound/tun[0]: connection closed: dial tcp [2a01:111:2003::52]:80: i/o timeout
DEBUG[0115] [2962870022] inbound/tun[0]: connection closed: io: read/write on closed pipe | upstream: context canceled
DEBUG[0115] [3843120482] inbound/tun[0]: connection closed: io: read/write on closed pipe | upstream: context canceled
DEBUG[0115] [927781168] inbound/tun[0]: connection closed: io: read/write on closed pipe | upstream: context canceled
DEBUG[0115] [2748979187] inbound/tun[0]: connection closed: io: read/write on closed pipe | upstream: context canceled
DEBUG[0115] [4082274589] inbound/tun[0]: connection closed: dial tcp [240e:97d:2014:204::6]:443: i/o timeout
```

</details>

**Comment by ActiveIce** [comment]
Fix in 1.1-beta15

**Comment by lyc8503** [comment]
在 v1.1 版本遇到这个问题, 发现是 TUN 的 inbound 中忘记加入 "inet6_address": "fd0a::/126" 导致的, 加入后就没有问题了.
